### PR TITLE
[terra-button] Show focus styles when programmatically focused

### LINKED ISSUE #2525: [terra-button] Update visual behavior to handle programmatic focus
# Feature Request
Update visual style for the programmatic setting of focus.

## Description
When setting focus on a terra-button through javascript default to keyboard style focus to ensure when navigating between elements with other keys than tab.

This relates back to the implementation of menus within the terra-application-navigation, where keyboard navigation uses arrow keys as well as the tab key.

## @ Mentions
@scottwilmarth

### PR DESCRIPTION
**Pull request by Blackop778**
### Summary
`Button` will now show focus styles when programmatically focused. I'm not really sure why it uses a focus state and `is-focused` class but I'm leaving that alone.

### Additional Details
Closes #2525

### CODE REVIEW

File: packages/terra-button/src/_mixins.scss [hunk 1]
```diff
@@ -8,7 +8,8 @@
   color: map-get($color-map, color);
   font-weight: map-get($color-map, font-weight);
 
-  &.is-focused {
+  &.is-focused,
+  &:focus {
```
Comment (dkasper-was-taken) [inline packages/terra-button/src/_mixins.scss]: Solving the problem this way will force mouse selection to also get focus, which we don't want, so we'll need to try solving this another way.

File: packages/terra-button/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Fixed
+* Fixed Button not showing focus styles when programmatically focused
```
Comment (emilyrohrbough) [inline packages/terra-button/CHANGELOG.md]: What do you mean by programmatically focused?

Comment (Blackop778) [reply packages/terra-button/CHANGELOG.md]: Programmatic focus would be getting a reference to the DOM element and calling the `focus()` method.

**Comment by emilyrohrbough** [comment]
>  I'm not really sure why it uses a focus state and is-focused class but I'm leaving that alone.

This is to allow keyboard focus styles to be applied to the button consistently across browsers: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus

I wish I had the original docs I found on this, but that was a while ago

**Comment by Blackop778** [comment]
> > I'm not really sure why it uses a focus state and is-focused class but I'm leaving that alone.
> 
> This is to allow keyboard focus styles to be applied to the button consistently across browsers: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
> 
> I wish I had the original docs I found on this, but that was a while ago

I guess it makes more sense if we don't want the button to display focus styles when it gets clicked. Don't think I've seen that documented anywhere though.

File: packages/terra-button/tests/wdio/button-spec.js [hunk 3]
```diff
@@ -238,4 +238,18 @@ Terra.describeViewports('Button', ['tiny'], () => {
...
     Terra.it.validatesElement({ selector: '#compactButton' });
   });
+
+  describe('Programmatic Focus', () => {
+    before(() => {
+      browser.url('/#/raw/tests/terra-button/button/programmatic-focus');
+      browser.click('#button1');
+    });
+
+    Terra.it.matchesScreenshot('Button 2 focused programmatically');
```
Comment (emilyrohrbough) [inline packages/terra-button/tests/wdio/button-spec.js]: Shouldn't this be button 1?

Comment (Blackop778) [reply packages/terra-button/tests/wdio/button-spec.js]: button 1's callback focuses button 2 and vice versa. Button 1 was just clicked so button 2 is now focused.

File: packages/terra-button/src/_mixins.scss (hunk 1 above)
Comment (Blackop778) [reply packages/terra-button/src/_mixins.scss]: Fixed in 06a47fb35b2e4d69e5204e17572a246f8b1357a6

**Pull request merged**